### PR TITLE
Clarify definition of --pull options

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/buildah/issues/5406

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: docs/buildah-build.1.md
Comment (TomSweeneyRedHat) [inline docs/buildah-build.1.md:750]: this doesn't specify what happens with true or false, does it?

**Comment by TomSweeneyRedHat** [comment]
rebase needed.  I think similar changes are needed to buildah-commit.1.md

File: docs/buildah-build.1.md
Comment (rhatdan) [reply docs/buildah-build.1.md:754]: No we want to hide those, since we do not want people using them any longer.  Just use the newer terminology so it is consistent with Podman.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat since buildah commit is pulling SBOM Scanner image, I will leave this to @nalind to decide, lets get this merged so that standard images work in the same was as podman.

**Comment by TomSweeneyRedHat** [comment]
@rhatdan all kinds or red test unhappiness here.

File: buildah.go
Comment (nalind) [inline buildah.go:65]: API break.

File: define/pull.go
Comment (nalind) [inline define/pull.go:50]: API break.

File: define/build.go
Comment (nalind) [inline define/build.go:124]: This changes the default (unset) value from `PullIfMissing` to `config.PullPolicyAlways`.

File: docs/buildah-build.1.md
Comment (nalind) [reply docs/buildah-build.1.md:754]: This drops all mention of SBOM scanner images.

File: cmd/buildah/from.go
Comment (krrg) [inline cmd/buildah/from.go:76]: Possibly this should be `ifnewer` and not `newer`? [Related PR that fixes the docs](https://github.com/containers/buildah/pull/5524).

Comment (rhatdan) [reply cmd/buildah/from.go:76]: I would rather go with Podman's definitions so these matchup.

File: tests/bud.bats
Comment (nalind) [inline tests/bud.bats:4260]: This was previously going to _have_ to pull the image in order for the build to complete at all, but the newly-inserted `--pull=missing` run will make that unnecessary, so it either needs to specify a different base image, or it needs to be removed in between.

File: cmd/buildah/from.go
Comment (nalind) [inline cmd/buildah/from.go:72]: The commit log and PR description don't mention this change in the default.  It seems important.

File: tests/from.bats
Comment (rhatdan) [reply tests/from.bats:82]: No I could put it back and maybe test both pull=false and pull=never.

File: pkg/cli/common.go
Comment (nalind) [inline pkg/cli/common.go:269]: This looks like it forgot "newer"

File: cmd/buildah/from.go
Comment (nalind) [reply cmd/buildah/from.go:76]: The description of "newer" looks like it's missing a word between "the" and "in"

**Comment by nalind** [comment]
LGTM

**Comment by rhatdan** [comment]
@TomSweeneyRedHat @flouthoc @giuseppe @mheon PTAL

**Review by flouthoc — APPROVED** [review]
LGTM
/lgtm
/hold

**Pull request merged**
